Improve typings

JUVOJustin opened the pull request:
I added more typings supported with php 7.4 as minimum. Also suppressed a warning in the VehicleQueryIterator

Commit:
Refactor Query and VehicleQuery methods to improve typing.

Updated method signatures, added nullable and strict type hints, and enhanced docblocks for better clarity and type safety. Ensured all return types and parameters align with modern PHP standards.

Commit:
Normalize indentation and formatting in Query classes

Standardized the indentation and formatting of the `Query` and `VehicleQuery` classes for improved readability and consistency. This includes fixing inconsistent spacing, aligning method blocks, and adjusting inline comments where necessary.

Commit:
Merge remote-tracking branch 'origin/master' into improve-typings

# Conflicts:
#	src/VehicleQuery.php

Commit:
Refactor type hints and improve docstrings in VehicleQuery

Updated method signatures to use strict type hints for better consistency and type safety. Enhanced and corrected docstrings for improved clarity, fixing typos and providing detailed parameter/return descriptions.

Commit:
Add ReturnTypeWillChange attribute to key() method

This change ensures compatibility with PHP 8.1+ by adding the ReturnTypeWillChange attribute to suppress deprecation warnings. Future adjustments to method signature may still be required based on PHP updates.

nadar commented:
sure, since we don't recommend autoscout to anyone, feel free to send prs.

nadar commented on src/Base/Query.php:
```diff
@@ -6,20 +6,27 @@
+    /**
+     * Retrieves the client instance associated with the object.
+     *
+     * @return Client|null The client instance if set, or null if not set.
```
please remove @return phpdoc if you define return type hinting

JUVOJustin replied:
I updated the comment to `@return self`. To make it more clear. 
is there a general issue with these phpdocs? https://docs.phpdoc.org/guide/references/phpdoc/tags/return.html states the commend is recommend everywhere except for constructors or void return type.

nadar replied:
you can remove the phpdoc @return since the method return type hinting is defined. `public function getClient(): ?Client` either use phpdoc or use method return type hint (which is of course the better solution)

JUVOJustin replied:
The comment is not only there to indicate typing, which in this case is indeed redundant to the typing of the function, but also to describe what is being returned and in which case.

I think it is useful in general, but the decision is up to you. Happy to adjust to your code style :)

JUVOJustin replied:
@nadar sorry for being the annoying one, but any feedback? :)

Commit:
Enhance date parsing and handle potential null values

Updated the `dateParser` method signature to enforce type safety and adjust its return type. Improved date-related methods to handle missing data gracefully by providing default empty strings. These changes increase code robustness and prevent potential runtime errors.

Commit:
Update return type in PHPDoc from $this to self

Aligned the PHPDoc return type with the actual method return type (`self`) for better clarity and accuracy. This improves code readability and documentation consistency.

JUVOJustin commented:
> sure, since we don't recommend autoscout to anyone, feel free to send prs.

I get why, but they are still a major player and clients of ours still use them :) So whenever we encounter issues we will try to contribute fixes.

nadar commented on src/VehicleQuery.php:
```diff
@@ -259,29 +269,31 @@ public function findAll()
     {
-        $response = $this->getClient()->endpointResponse('vehicles/'.$id, $this->_where);
-        
+        $response = $this->getClient()->endpointResponse('vehicles/' . $id);
```
@JUVOJustin $this->_where has been removed. why?

JUVOJustin replied:
Good catch! Sorry for the inconvenience.

Commit:
Update findOne method to include query filters again

Modified the `findOne` method to pass additional query filters via the `_where` property. This ensures the endpoint response reflects any applied conditions for more precise results.

nadar commented:
@JUVOJustin https://github.com/nadar/autoscout24/releases/tag/1.4.0
